### PR TITLE
Feature/product search endpoint

### PR DESCRIPTION
**Pull request by rhimmelbauer**
Notes:

- Created a TableFilterMixn to handle pagination, ordering and filtering.
- Updated products.html to send filtering, ordering and pagination.
- AdminProductListView implements TableFilterMixin

### CODE REVIEW

File: vendor/templates/vendor/manage/products.html [hunk 1]
```diff
@@ -1,50 +1,185 @@
...
+          {% endfor %}
+        </tbody>
+      </table>
+    </div>
+
+
+    <div class="pagination justify-content-between">
+      <span>
+        {% blocktrans with object_list|length as total_entries %}
+        Showing {{page_obj.number}} to {{page_obj.number + page_obj.paginate_by }} of {{total_entries}}
```
Comment (mgcamisa) [inline vendor/templates/vendor/manage/products.html:85]: you should be able to use page_obj.start_index and end_index for this part

File: vendor/templates/vendor/manage/products.html [hunk 2]
```diff
@@ -1,50 +1,185 @@
...
+        {% endblocktrans %}
+      </span>
+      <span>
+        {% if page_obj.has_previous %}
+        <a href="?page=1">&laquo; first</a>
```
Comment (mgcamisa) [inline vendor/templates/vendor/manage/products.html:90]: this section needs translations when you get the chance

File: vendor/templates/vendor/manage/products.html [hunk 3]
```diff
@@ -1,50 +1,185 @@
...
+        $(location).attr('href', getFilterKWARG());
+      }
+    })
+
+    $("#search_button").on("click", function (event) {
+      $(location).attr('href', getFilterKWARG());
+    })
+
+    $("#product_order").on("click", function (event) {
+      console.log("#product_order");
```
Comment (mgcamisa) [inline vendor/templates/vendor/manage/products.html:152]: should probably remove console.log

File: vendor/templates/vendor/manage/products.html (hunk 1 above)
Comment (rhimmelbauer) [reply vendor/templates/vendor/manage/products.html:85]: changed thanks

File: vendor/templates/vendor/manage/products.html (hunk 2 above)
Comment (rhimmelbauer) [reply vendor/templates/vendor/manage/products.html:90]: trans added

File: vendor/templates/vendor/manage/products.html (hunk 3 above)
Comment (rhimmelbauer) [reply vendor/templates/vendor/manage/products.html:152]: removed

**Review by mgcamisa — APPROVED** [review]

**Pull request merged**